src/amr/messengers/hybrid_hybrid_messenger_strategy.hpp

```diff
@@ -841,11 +796,45 @@ namespace amr
                    / (afterPushCoarseTime_[levelNumber] - beforePushCoarseTime_[levelNumber]);
         }
 
+
+        void debug_print(VecFieldT const& B, GridLayoutT const& layout, int loc, int ix, int iy,
+                         std::string const& aftbef)
+        {
+            auto& Bx       = B(core::Component::X);
+            auto& By       = B(core::Component::Y);
+            auto& Bz       = B(core::Component::Z);
+            auto const& dx = layout.meshSize()[0];
+            auto const& dy = layout.meshSize()[1];
+
+            if (loc == 3) // w hi, y hi
+            {
+                std::cout << aftbef << "\n";
```

Comment:
maybe this could be a debug build only thing?

multiple ways to do that, either use `PHARE_LOG_LINE_STR`

or you could inject this at the start of the function
```
#if defined(NDEBUG)
return;
#endif
```

Reply:
```cpp
PHARE_DEBUG_DO(const std::string before = "BEFORE";
debug_print(B, layout, loc, ix, iy, before);)
```
?

Reply:
that should also work